Source/Projections/Builder/Copies/MongoDB/ConversionsFromBsonClassMapAdder.cs

```diff
@@ -0,0 +1,110 @@
+        else if (TryResolveConversionFromSerializer(serializer, out var conversion))
+        {
+            conversions.AddConversion(path, conversion);
+        }
+    }
+    static bool TryResolveConversionFromSerializer(IBsonSerializer serializer, out Conversion conversion)
+    {
+        conversion = Conversion.None;
+        if (serializer is IBsonArraySerializer arraySerializer && arraySerializer.TryGetItemSerializationInfo(out var info))
+        {
+            serializer = info.Serializer;
+        }
+        switch (serializer)
+        {
+            case GuidSerializer guidSerializer:
+                if (guidSerializer.Representation == BsonType.String)
+                {
+                    conversion = Conversion.GuidAsString;
+                    return true;
+                }
+                else if (guidSerializer.Representation == BsonType.Binary)
+                {
+                    switch (guidSerializer.GuidRepresentation)
+                    {
+                        case GuidRepresentation.Standard:
+                            conversion = Conversion.GuidAsStandardBinary;
+                            return true;
+                        case GuidRepresentation.CSharpLegacy:
+                        case GuidRepresentation.Unspecified:
+                            conversion = Conversion.GuidAsCSharpLegacyBinary;
+                            return true;
+                    }
+                }
+                return false;
+            case DateTimeSerializer dateTimeSerializer:
+                return TryGetConversionFromDateBsonRepresentation(dateTimeSerializer.Representation, out conversion);
+                
+            case DateTimeOffsetSerializer dateTimeOffsetSerializer:
+                return TryGetConversionFromDateBsonRepresentation(dateTimeOffsetSerializer.Representation, out conversion);
+            default:
+                return false;
+        }
+    }
+
+    static bool TryGetConversionFromDateBsonRepresentation(BsonType representation, out Conversion conversion)
+    {
+        conversion = Conversion.None;
+        switch (representation)
+        {
+            case BsonType.Array:
+                conversion = Conversion.DateAsArray;
+                return true;
+            case BsonType.Document:
+                conversion = Conversion.DateAsDocument;
+                return true;
+            case BsonType.DateTime:
+                conversion = Conversion.DateAsDate;
+                return true;
+            case BsonType.String:
+                conversion = Conversion.DateAsString;
+                return true;
+            case BsonType.Int64:
+                conversion = Conversion.DateAsInt64;
+                return true;
+            default:
+                return false;
+        }
+    }
+    static bool IsComplexType(Type nominalType, IBsonSerializer serializer) => typeof(BsonClassMapSerializer<>).MakeGenericType(nominalType).IsInstanceOfType(serializer);
```

Comment:
Not sure if this is a problem, but maybe it is better to check if the `serializer` is an instance of `BsonClassMapSerializer<>` without the generic type specified? I think you can do something with `nominalType.GetGenericType()`. Just thinking about inheritance and serialisers for base classes.